contracts/TokenIOLib.sol

```diff
@@ -1222,5 +1209,24 @@ library TokenIOLib {
     return usdAmount;
   }
 
+  function compare(string _a, string _b) internal pure returns (bool) {
+        bytes memory a = bytes(_a);
+        bytes memory b = bytes(_b);
+        uint minLength = a.length;
+        if (b.length < minLength) minLength = b.length;
+        //@todo unroll the loop into increments of 32 and do full 32 byte comparisons
+        for (uint i = 0; i < minLength; i ++)
+            if (a[i] < b[i])
+                return false;
+            else if (a[i] > b[i])
+                return false;
+        if (a.length < b.length)
```

Comment:
length should be checked before the loop and can be done with one comparison, no?
the comparison doesn't look effective in general

Reply:
Fixed